src/kernel/capinv.c

```diff
@@ -624,15 +602,16 @@ static int
 cap_thd_op(struct cap_thd *thd_cap, struct thread *thd, struct pt_regs *regs,
 	   struct comp_info *ci, struct cos_cpu_local_info *cos_info)
 {
-	struct thread *next = thd_cap->t;
-	capid_t arcv        = (__userregs_get1(regs) << 16) >> 16;
-	capid_t tc          = __userregs_get1(regs) >> 16;
-	u32_t prio_higher   = __userregs_get3(regs);
-	u32_t prio_lower    = __userregs_get2(regs);
-	tcap_prio_t prio    = (tcap_prio_t)(prio_higher >> 16) << 32 | (tcap_prio_t)prio_lower;
-	sched_tok_t usr_counter   = (((sched_tok_t)__userregs_get3(regs) << 16) >> 16) | ((sched_tok_t)__userregs_getop(regs) << 16); /* op holds MSB of counter */
-	tcap_time_t timeout = (tcap_time_t)__userregs_get4(regs);
-	struct tcap *tcap   = tcap_current(cos_info);
+	struct thread *next       = thd_cap->t;
+	capid_t arcv              = (__userregs_get1(regs) << 16) >> 16;
+	capid_t tc                = __userregs_get1(regs) >> 16;
+	u32_t prio_higher         = __userregs_get3(regs);
+	u32_t prio_lower          = __userregs_get2(regs);
+	tcap_prio_t prio          = (tcap_prio_t)(prio_higher >> 16) << 32 | (tcap_prio_t)prio_lower;
+	sched_tok_t usr_counter   = (((sched_tok_t)__userregs_get3(regs) << 16) >> 16) 
```

Comment:
the argument for doing it this ways is that the # of bits removed is easier to discern from `>> <<` than a mask (people don't think in hex).  Neither is correct; that is the argument.

The "correct" way is to use a macro/function, but we're customizing each system call so much that even this would be a mess.